### DISCUSSION TITLE
forge coverage discrepancy in Foundry Fundamentals / Section 2 Foundry Fund Me

### DISCUSSION THREAD

**Discussion by auditoor** [Cyfrin/foundry-full-course-cu#2218] [Q&A] [answered]
Hello auditoors!

I am having an issue when I run forge coverage at the beginning of Chapter 14 Section 2 (Foundry Fund Me). 

How am I getting 100% coverage for deployfundme.s.sol while Patrick gets 0%?

<img width="952" alt="Screenshot 2024-08-11 at 11 55 57 AM" src="https://github.com/user-attachments/assets/89edbe7f-63f1-458f-bafb-549deb5ef9cb">

This is what my deployfundme.s.sol looks like:

```solidity
// SPDX-License-Identifier: MIT
pragma solidity 0.8.19;

import {Script} from "forge-std/Script.sol";
import {fundme} from "../src/fundme.sol";
import {helperconfig} from "../script/helperconfig.s.sol";

contract deployfundme is Script {
    function run() external returns (fundme) {
        helperconfig Helperconfig = new helperconfig();
        address ethusdpricefeed = Helperconfig.config();

        vm.startBroadcast();
        fundme Fundme = new fundme(ethusdpricefeed);
        vm.stopBroadcast();
        return Fundme;
    }
}
```

And here is my fundmetest.t.sol:

```solidity
// SPDX-License-Identifier: MIT
pragma solidity 0.8.19;

import {Test, console} from "forge-std/Test.sol";
import {fundme} from "../src/fundme.sol";
import {deployfundme} from "../script/deployfundme.s.sol";

// import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
// import {priceconverter} from "src/priceconverter.sol";

contract fundmetest is Test {
    fundme Fundme;

    function setUp() external {
        deployfundme Deployfundme = new deployfundme();
        Fundme = Deployfundme.run();
    }

    function testminimumusdisfive() public view {
        assertEq(Fundme.MINIMUM_USD(), 5e18);
    }

    function testownerismsgsender() public view {
        // console.log("msg.sender is", msg.sender);
        // console.log("i_owner is", Fundme.i_owner());
        // console.log("The test contract address is", address(this));

        assertEq(Fundme.i_owner(), msg.sender);
        // The test contract deploys the Fundme contract inside the setUp function and the i_owner is set to the address of the contract.
    }

    function testpricefeedversionisaccurate() public view {
        uint256 version = Fundme.getVersion();
        // console.log("version is", version);
        assertEq(version, 4);
    }
}
// // We could make this constant?  /* hint: no! We should make it immutable! */
// address public /* immutable */ i_owner;
// uint256 public constant MINIMUM_USD = 5 * 10 ** 18;

```

Thank you for your time!

**Comment by EngrPips** [comment]
Hello @auditoor, I think you are getting 100% coverage because you applied your `deployfundme` script inside of your test setup 

```js
function setUp() external {
        deployfundme Deployfundme = new deployfundme();
        Fundme = Deployfundme.run(); <@ this line
    }

```
Did Patrick did the same in the course?

**Reply by auditoor** [reply]
Thank you @EngrPips!

I just went back to check chapter 13 and it seems Patrick is also running the `deployfundme` script inside the test file.
<img width="1440" alt="Screenshot 2024-08-11 at 5 26 23 PM" src="https://github.com/user-attachments/assets/dd7bfba4-7fcb-4857-b449-57cc4d3fffb1">

**Reply by EngrPips** [reply]
Interesting, he ran the script in his test as well but got `0` coverage for the `deployfundme` script?

**Reply by auditoor** [reply]
Yes. I'm sure I am missing something here but it looks like the forge coverage discrepancy doesn't come from the fact that he runs the deploy script inside the test file.

**Reply by EngrPips** [reply, marked as answer]
I am unsure what the reason for that discrepancy is, but then maybe `Foundry` way of calculating coverage recently included anything processed inside of the test `setUp` function too, which wasn't the case before. I didn't read this from any official documentation, but I am just trying to make sense of this situation.

**Reply by auditoor** [reply]
I found this after a really quick search. Could this be the explanation?

https://github.com/foundry-rs/foundry/pull/6123

**Reply by EngrPips** [reply]
> I found this after a really quick search. Could this be the explanation?
> 
> [foundry-rs/foundry#6123](https://github.com/foundry-rs/foundry/pull/6123)

It seem like I was correct. Since the day I started making sense of things rather than understanding people's definition or explanations a lot of things have been making sense in my study, lol.
